java/test/src/main/java/io/ray/test/AnonymousNamespaceTest.java

```diff
@@ -0,0 +1,41 @@
+package io.ray.test;
+
+import io.ray.api.ActorHandle;
+import io.ray.api.Ray;
+import java.io.IOException;
+import java.util.NoSuchElementException;
+import java.util.concurrent.TimeUnit;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+@Test(groups = "cluster")
+public class AnonymousNamespaceTest {
+
+  private static class A {
+    public String hello() {
+      return "hello";
+    }
+  }
+
+  public void testIsolationBetweenAnonymousNamespaces() throws IOException, InterruptedException {
+    NamespaceTest.testIsolation(
+        AnonymousNamespaceTest.class,
+        () ->
+            Assert.assertThrows(
+                NoSuchElementException.class,
+                () -> {
+                  Ray.getGlobalActor("a").get();
+                }));
+  }
+
+  public static void main(String[] args) throws IOException, InterruptedException {
```

Comment:
Maybe you can create an inner class in `NamespaceTest` with `main` method, so you can put the test case into `NamespaceTest` and avoid adding a new test file.